kotlin/services/route53/src/main/kotlin/com/kotlin/route/Route53Scenario.kt

```diff
@@ -0,0 +1,317 @@
+        val response = route53DomainsClient.checkDomainTransferability(transferabilityRequest)
+        println("Transferability: ${response.transferability?.transferable}")
+    }
+}
+// snippet-end:[route.kotlin.checkdomaintransfer.main]
+
+// snippet-start:[route.kotlin.checkdomainavailability.main]
+suspend fun checkDomainAvailability(domainSuggestion: String) {
+    val availabilityRequest = CheckDomainAvailabilityRequest {
+        domainName = domainSuggestion
+    }
+    Route53DomainsClient { region = "us-east-1" }.use { route53DomainsClient ->
+        val response = route53DomainsClient.checkDomainAvailability(availabilityRequest)
+        println("$domainSuggestion is ${response.availability}")
+    }
+}
+// snippet-end:[route.kotlin.checkdomainavailability.main]
+
+// snippet-start:[route.kotlin.domainsuggestions.main]
+suspend fun listDomainSuggestions(domainSuggestion: String?) {
+    val suggestionsRequest = GetDomainSuggestionsRequest {
+        domainName = domainSuggestion
+        suggestionCount = 5
+        onlyAvailable = true
+    }
+    Route53DomainsClient { region = "us-east-1" }.use { route53DomainsClient ->
+        val response = route53DomainsClient.getDomainSuggestions(suggestionsRequest)
+        response.suggestionsList?.forEach { suggestion ->
+            println("Suggestion Name: ${suggestion.domainName}")
+            println("Availability: ${suggestion.availability}")
+            println(" ")
+        }
+    }
+}
+// snippet-end:[route.kotlin.domainsuggestions.main]
+
+// snippet-start:[route.kotlin.domainprices.main]
+suspend fun listAllPrices(domainType: String?) {
+    val pricesRequest = ListPricesRequest {
+        tld = domainType
+    }
+    Route53DomainsClient { region = "us-east-1" }.use { route53DomainsClient ->
+        val response = route53DomainsClient.listPrices(pricesRequest)
+        response.prices?.forEach { price ->
+            println("Name: ${price.name}")
+            println("Registration price: ${price.registrationPrice}")
+            println("Renewal: ${price.renewalPrice}")
+        }
+    }
+}
+// snippet-end:[route.kotlin.domainprices.main]
+
+// snippet-start:[route.kotlin.domainbillingrecords.main]
+suspend fun listBillingRecords() {
+    val currentDate = Date()
+    val localDateTime = currentDate.toInstant().atZone(ZoneId.systemDefault()).toLocalDateTime()
+    val zoneOffset = ZoneOffset.of("+01:00")
+    val localDateTime2 = localDateTime.minusYears(1)
+    val myStartTime = localDateTime2.toInstant(zoneOffset)
+    val myEndTime = localDateTime.toInstant(zoneOffset)
+    val timeStart: Instant? = myStartTime?.let { Instant(it) }
+    val timeEnd: Instant? = myEndTime?.let { Instant(it) }
+
+    val viewBillingRequest = ViewBillingRequest {
+        start = timeStart
+        end = timeEnd
+    }
+    Route53DomainsClient { region = "us-east-1" }.use { route53DomainsClient ->
+        val response = route53DomainsClient.viewBilling(viewBillingRequest)

```

Comment:
Was there a paginator for listing billing and listing operations? Like this: https://docs.aws.amazon.com/sdk-for-kotlin/latest/developer-guide/pagination.html ?